### DISCUSSION TITLE
How is impersonation supposed to be set up?

### DISCUSSION THREAD

**Discussion by marzlevane** [apache/airflow#20831] [Q&A] [answered]
I'm trying out the default_impersonation feature in order to separate kerberos credentials needed to access the Airflow metadata db (associated with the airflow user) from credentials needed to access business data dbs (associated with a taskuser).

I have started from the official v2.2.3/python3.9 docker image from docker hub and added a "taskuser". I also setup sudo privileges for the airflow user.

I then set the dafault_impersonation in airflow.cfg to taskuser. A test dag (with simple python operator task) executes with sudo -u taskuser, but fails as the command being run is "airflow tasks run", and command "airflow" cannot be found when executed as "taskuser".

I understand that airflow is installed only for user "airflow" in the official image, in ~airflow/.local/. How is the airflow installation supposed to be set up for impersonation to work as intended? If the task is run as a call to airflow, what privileges does the impersonated user need? Execute permissions on "airflow"? Access to any other airflow configuration?

I tried looking for examples and documentation, but all I found was the very short section that mentions that sudo is needed. I also saw that the impersonation implementation was altered in 2.2.3 to use sudo, does this alteration change how the airflow installation and permissions should be set up?

**Comment by potiuk** [comment, marked as answer]
You should setup home directory to be the same as airflow user (/home/airflow) and have it belong to 0 group. See https://airflow.apache.org/docs/docker-stack/entrypoint.html#allowing-arbitrary-user-to-run-the-container

**Reply by marzlevane** [reply]
Thanks for the help potiuk

Maybe this should be included in the documentation [here](https://airflow.apache.org/docs/apache-airflow/1.10.10/security.html?highlight=impersonation#impersonation)?

I also needed to add /home/airflow/.local/bin to the secure_path in /etc/sudoers for the setup to work.

**Reply by potiuk** [reply]
@marzlevane: By all means, feel free to update the docs! This is super easy. Just use "Suggest a change on this page" at the bottom right of the page and it will open Pull Request where you will be able to update the page. You are one of the best people in the world to update it as you struggled with it and you will know best how to put it in words for others like, so they can follow it.

It's an easy way to become one of more than 1900 contributors to Airlow this way !
